lib/services/cognitiveServicesSearch/README.md

```diff
@@ -0,0 +1,98 @@
+# Microsoft Azure SDK for Node.js - Cognitive Services Search
+
+This project provides a Node.js package that makes it easy to work with Microsoft Azure Cognitive Services Entity Search API. Right now it supports:
+- **Node.js version: 7.10.0 or higher**
+
+
+## How to Install
+
+```bash
+npm install azure-cognitiveservices-search
+```
+
+## How to use
+
+### Create a Cognitive Services Bing Search Account
+
+```javascript
+  const msRestAzure = require('ms-rest-azure');
+  const CognitiveServicesManagement = require("azure-arm-cognitiveservices");
+  let client;
+  let createAccount = msRestAzure.interactiveLogin().then((credentials) => {
+    client = new CognitiveServicesManagement(credentials, suite.subscriptionId);
+    return client.accounts.create('groupname', 'accountname', {
+      sku: {
+        name: "S1"
+      },
+      kind: "Bing.Search",
+      location: "westus",
+      properties: {}
+    });
+  }).catch((err) => {
+    console.log('An error ocurred');
+    console.dir(err, {depth: null, colors: true});
+  });
+```
+
+### List the keys from the created account
+
+```javascript
+  let serviceKey;
+  createAccount.then((result) => {
+    return client.accounts.listKeys('groupname', 'accountname');
+  }).then((result) => {
+    serviceKey = result.key1;
+    console.log(result.key2);
+  }).catch((err) => {
+    console.log('An error ocurred');
+    console.dir(err, {depth: null, colors: true});
+  });
+```
+
+### Create credentials
+
+ ```javascript
+ const CognitiveServicesCredentials = require('ms-rest-azure').CognitiveServicesCredentials;
+ let credentials = new CognitiveServicesCredentials(serviceKey);
+ ```
+
+### Query the Entity Search API
+
+ ```javascript
+const search = require('azure-cognitiveservices-search');
+
+let entitySearchApiClient = new search.EntitySearchAPIClient(credentials);
+let query = 'seahawks';
+entitySearchApiClient.entitiesOperations.search(query, null, function(err, result, request, response){
```

Comment:
Why are you passing `null` as the second parameter? Is it a required parameter or an optional parameter? Just letting you know that we need not pass null values for optional parameters, the method will internally autoshift the callback ahead by 1 is the optional parameter was not passed.

Reply:
It's optional. I had left the entity search example largely the way it was.
Will change.